core/conventional-commits/lib/update-changelog.js

```diff
@@ -66,8 +71,23 @@ function updateChangelog(pkg, type, { changelogPreset, rootPath, tagPrefix = "v"
       const content = [CHANGELOG_HEADER, newEntry, changelogContents].join(BLANK_LINE);
 
       return fs.writeFile(changelogFileLoc, content.trim() + EOL).then(() => {
-        log.verbose(type, "wrote", changelogFileLoc);
+        // if change log preview option is passed open the file in vim before commiting
+        if (changelogPreview) {
+          return new Promise((resolve, reject) => {
+            const vim = spawn("vim", [changelogFileLoc], { stdio: "inherit" });
```

Comment:
This is a bit specific, what if I don't like `vim`, and prefer `code` instead? What if my platform doesn't have `vim`? (i mean, yeah, "don't use `--changelog-preview`" is a reasonable answer to the latter question)